src/core/model.js

```diff
@@ -429,6 +429,16 @@ in a future version. Use shader modules instead.`);
     });
   }
 
+  _recycleResources() {
+    // Add program object to cache
+    if (this.shaderCache) {
+      this.program.resetState();
```

Comment:
* The resetState call should be done inside shaderCache.addProgram(), not here. 
* Can we set the shaderCache on the Program, then we just do `program.delete()` and the program updates the cache?
* That could help address the issue that if the app supplies a program, it could be shared (by the app) and then shouldn't be reused/deleted by the model. Only programs created by Model should be reused/deleted?

Reply:
> That could help address the issue that if the app supplies a program, it could be shared (by the app) and then shouldn't be reused/deleted by the model. Only programs created by Model should be reused/deleted?

Yes that sounds like a good direction we should go, giving more control the app on how to recycle its resources. I assume in those cases, app doesn't provide 'shaderCache` so they never get added or may be some other means like overwriting delete method?